scripts/publish.ts

```diff
@@ -0,0 +1,148 @@
+import { readdirSync, readFileSync, writeFileSync } from 'fs'
+import { join, resolve } from 'path'
+import { chdir } from 'process'
+import { prompt, Separator } from 'inquirer'
+import * as spawn from 'await-spawn'
+import { blue, green, red } from 'colors'
+
+const readJSON = (jsonPath: string): any => {
+  return JSON.parse(
+    readFileSync(jsonPath, { encoding: 'utf-8' as BufferEncoding })
+  )
+}
+
+const writeJSON = (jsonPath: string, object: any): void => {
+  writeFileSync(jsonPath, JSON.stringify(object, null, 2) + '\n')
+}
+
+const fromEntries = (xs: [string | number | symbol, any][]) =>
+  xs.reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {})
+
+const spawnProcess = async (
+  command: string,
+  args: string[] = [],
+  log?: { onSuccess?: any }
```

Comment:
change any to () => void

Reply:
most times it's better to abort after a process fails to avoid mixing up too many error messages (you can leave the console.error and rethrow)
I suggest getting rid of the onSuccess and just directly calling what you calls from onSuccess after the spawnProcess (since you can always do await).
Another option would be that you pass a successMessage to spawnProcess, so that you don't need to repeat "console.log(greeen" so often